lib/S3Plugin.js

```diff
@@ -105,8 +105,15 @@ class S3Plugin {
     }
 
     this.context = context;
-
-    this.baseFileUrl = `https://s3.${this.config.region}.amazonaws.com/${this.config.bucketName}`;
+    //Build the endpoint parameter if we only have the region
+    if (this.config.endpoint === 'undefined') {
+      this.config.endpoint = `https://s3.${this.config.region}.amazonaws.com`;
+      this.config.s3ForcePathStyle = false;
```

Comment:
I don't understand, what this variable is used for ?

Reply:
I agree about the endpoint being the only parameter.
The s3ForcePathStyle is a s3 config parameter that is needed on some S3 based services when they use paths style instead of subdomains.
It is documented on the s3 api.
If we use the endpoint aproach we will need to expose it  as an optional parameter to the configuration, or defaulting to false.
```
{
  "plugins": {
    "s3": {
      // S3 bucket
      "bucketName": "your-s3-bucket",
      // S3 endpoint
      "endpoint": "https://<your-s3-compatible-endpoint-url>",
     // True if your s3-compatible service use paths style instead of subdomains
      "s3ForcePathStyle": false,
      // TTL in ms before Presigned URL expire or the uploaded file is deleted
      "signedUrlTTL": 1200000,
      // Redis key prefix
      "redisPrefix": "s3Plugin/uploads",
    }
  }
}
```

Reply:
Ok I see.
I'm thinking that it may be a good idea to allow users to pass also arbitrary options to the S3 client constructor. We could expose these options in another object and defaulting a `s3ForcePathStyle` inside to `false`.
What do you think?
```
{
  "plugins": {
    "s3": {
      // S3 bucket
      "bucketName": "your-s3-bucket",
      // S3 endpoint
      "endpoint": "https://<your-s3-compatible-endpoint-url>",
      // Additional options passed to the S3 client constructor
      "s3ClientOptions": {
         // True if your s3-compatible service use paths style instead of subdomains
        "s3ForcePathStyle": false,
      },
      // TTL in ms before Presigned URL expire or the uploaded file is deleted
      "signedUrlTTL": 1200000,
      // Redis key prefix
      "redisPrefix": "s3Plugin/uploads",
    }
  }
}
``